src/components/container/AIAssistant/AIAssistant.tsx

```diff
@@ -0,0 +1,120 @@
+            assistantRef.current = new MyDataHelpsAssistant(props.additionalInstructions, props.tools, props.appendTools);
+        }
+    }, []);
+
+    const addUserMessage = async function (newMessage: string) {
+
+        setMessages(prevMessages => [...prevMessages, { type: 'user', content: newMessage }]);
+
+        await assistantRef.current?.ask(newMessage, function (streamEvent: StreamEvent) {
+
+            const [kind, type] = streamEvent.event.split("_").slice(1);
+
+            if (type === "stream" && kind !== "chain") {
+                const chunk = streamEvent.data?.chunk;
+                let msg = chunk.message as AIMessageChunk;
+
+                if (msg.content && typeof msg.content === "string") {
+                    addMessageChunk(streamEvent.run_id, msg.content);
+                }
+                else if (props.debug && msg.tool_call_chunks && msg.tool_call_chunks.length > 0) {
+                    if (msg.tool_call_chunks[0].args) {
+                        addMessageChunk(streamEvent.run_id, msg.tool_call_chunks[0].args);
+                    }
+                    else if (msg.tool_call_chunks[0].name) {
+                        addMessageChunk(streamEvent.run_id, msg.tool_call_chunks[0].name + " ");
+                    }
+                }
+            }
+
+            if (kind === "tool") {
+                if (type === "start") {
+                    setLoading(`Querying your data...`);
+                }
+                else if (type === "end") {
+                    setLoading("");
+                }
+            }
+
+            if (kind === "llm" && type === "end") {
+                MyDataHelps.trackCustomEvent({
+                    eventType: "aiassistant-message",
+                    properties: {
+                        type: "ai",
+                        body: messages[messages.length - 1].content
+                    }
+                });
+            }
+        });
+
+        MyDataHelps.trackCustomEvent({
+            eventType: "aiassistant-message",
+            properties: {
+                type: "user",
+                body: newMessage
+            }
+        });
+    }
+
+    const addMessageChunk = function (runId: string, message: string) {
+        setMessages(prevMessages => {
+            let existingMessage = prevMessages.find((msg) => msg.runId === runId);
+            if (existingMessage) {
+                const updatedMessage = { ...existingMessage, content: existingMessage.content + message };
+                return prevMessages.map(msg => msg.runId === runId ? updatedMessage : msg);
+            }
+            else {
+                return [...prevMessages, { type: 'ai', content: message, runId: runId }];
+            }
+        });
+    }
+
+    return <>
+        {messages && <Chat innerRef={props.innerRef} messages={messages.map((msg) => {
+            return {
+                icon: msg.type === "ai" ? <FontAwesomeSvgIcon icon={faFlask} width={16} /> : undefined,
```

Comment:
Does the flask icon work for now? I kind of liked the flask with gear icon https://fontawesome.com/icons/flask-gear?s=solid but it's one of the paid fontawesome icons.

I know we discussed getting rid of icons completely but I have to admit things looked strange without the ai icon at all. It becomes a little unclear when the ai started a new response etc since it can talk in paragraphs.

Reply:
Ah I see - you could just go with a dot/bullet maybe if the goal is just to ensure you understand where it started - or an arrow or something

Reply:
I don't really love the flask because we use that for labs etc. and it doesn't necessarily have any significance to the consumer with an AI chatbot...like it feels like a flag that this is "experimental" perhaps but other than that i'm not sure it makes sense